libraries/AP_Proximity/AP_Proximity.cpp

```diff
@@ -198,6 +199,12 @@ void AP_Proximity::init()
             num_instances = instance+1;
         break;
 #endif
+#if AP_PROXIMITY_SCRIPTING_ENABLED
+        case Type::Lua_Scripting:
+            state[instance].instance = instance;
+            drivers[instance] = new AP_Proximity_Lua(*this, state[instance], params[instance]);
```

Comment:
nitpick: we'd normally name the backend AP_Proximity_Scripting() as well.  Again not a blocker.